conftest.py

```diff
@@ -644,16 +648,18 @@ def integration_lifecycle(svc_client, mock_redis):
 
     yield svc_client, headers, project_id, url_components
 
-    # Teardown step: Delete all branches except master.
-    with integration_repo(headers, url_components) as repo:
-        for repo_branch in repo.references:
-            if repo_branch.name == 'master':
-                continue
-
-            try:
-                repo.remote().push(refspec=(':{0}'.format(repo_branch.name)))
-            except git.exc.GitCommandError:
-                continue
+    # Teardown step: Delete all branches except master (if needed).
+    if integration_repo_path(headers, url_components).exists():
+        with integration_repo(headers, url_components) as repo:
+            for repo_branch in repo.references:
+                if repo_branch.name == 'master':
```

Comment:
Are we sure that `master` branch always exists? What if user changes the name of this branch?

Reply:
This code is already on master. Not sure why git is showing that I'm adding this ... maybe something went wrong with rebase. 

Since this is for solely for our integration tests I would say it's sufficient solution, since we have total control over how the repo looks like.